ui/admin-portal/src/app/components/opportunity/edit-opp/edit-opp.component.ts

```diff
@@ -62,7 +80,42 @@ export class EditOppComponent implements OnInit {
       nextStep: this.nextStep,
       nextStepDueDate: this.nextStepDueDate,
     }
-    this.activeModal.close(info)
+    const parentJobInfo: OpportunityProgressParams = {
+      nextStep: this.nextStep,
+      nextStepDueDate: this.nextStepDueDate,
+    }
+    if (this.copyToParentJob && isCandidateOpportunity(this.opp)) {
+      this.jobService.update(this.opp.jobOpp.id, parentJobInfo).subscribe(() => {
+          this.activeModal.close(info);
+        },
+        (error) => this.error = error);
+      } else {
+        this.activeModal.close(info)
+    }
   }
 
+  /**
+   * If opp is the only open case of its parent job, sets isOnlyOpenCaseOfParentJob to true — the job has a ShortSavedList property but checking for open cases requires the full SavedList, so this is retrieved before proceeding.
+   * @param opp CandidateOpportunity/case whose progress is being updated
+   *
+   */
+  setIsOnlyOpenCaseOfParentJob(opp: CandidateOpportunity) {
```

Comment:
The naming of this method should really change to simply "isOnlyOpenCaseOfParentJob". There is no need for the "set".

See for example https://en.wikipedia.org/wiki/JavaBeans for the naming conventions - using the "get", "set" and "is prefixes.

This is not really a "bean" - but it does some processing and then returns a boolean value, so a simple prefix of "is" would be logical.